android/src/main/java/com/tailscale/ipn/ui/notifier/Notifier.kt

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package com.tailscale.ipn.ui.notifier
+
+import android.util.Log
+import com.tailscale.ipn.ui.model.Ipn.Notify
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+typealias NotifierCallback = (Notify) -> Unit
+
+
+class Watcher(
+    val sessionId: String,
+    val mask: Int,
+    val callback: NotifierCallback
+)
+
+// Notifier is a wrapper around the IPN Bus notifier.  It provides a way to watch the
+// for changes in various parts of the Tailscale engine.  You will typically only use
```

Comment:
is there ever a case where you might have more than one Notifier?

Reply:
Yup.  Different notifiers can give you different streams...  It's probably not the case that we will have more than one, but I wanted to write it so we could.